iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/TsFileTableSchemaUtil.java

```diff
@@ -0,0 +1,178 @@
+      this.predicate = predicate;
+    }
+
+    public boolean test(final TsTableColumnCategory category) {
+      return predicate.test(category);
+    }
+  }
+
+  /**
+   * High-performance method to convert TsTable ByteBuffer to TableSchema with column filtering.
+   * Only parses and includes columns that match the filter, skipping others for efficiency. This is
+   * critical for tables with many columns (e.g., 10,000+ columns).
+   *
+   * @param tsTableBuffer ByteBuffer containing serialized TsTable
+   * @param filter Column category filter to determine which columns to include
+   * @return TableSchema object
+   */
+  private static TableSchema tsTableBufferToTableSchemaInternal(
+      final ByteBuffer tsTableBuffer, final ColumnCategoryFilter filter) {
+    final String tableName = ReadWriteIOUtils.readString(tsTableBuffer);
+    final int columnNum = ReadWriteIOUtils.readInt(tsTableBuffer);
+
+    final List<IMeasurementSchema> measurementSchemas = new ArrayList<>(columnNum);
+    final List<ColumnCategory> columnTypes = new ArrayList<>(columnNum);
+
+    for (int i = 0; i < columnNum; i++) {
+      final byte categoryByte = ReadWriteIOUtils.readByte(tsTableBuffer);
+      final TsTableColumnCategory category = TsTableColumnCategory.deserialize(categoryByte);
+
+      // Early filtering: if we don't need this column, skip all its data
+      if (!filter.test(category)) {
+        skipColumnData(tsTableBuffer, category);
+        continue;
+      }
+
+      // Only parse data for columns we need
+      final String columnName = ReadWriteIOUtils.readString(tsTableBuffer);
+      final TSDataType dataType = ReadWriteIOUtils.readDataType(tsTableBuffer);
+
+      if (category == TsTableColumnCategory.FIELD) {
+        ReadWriteIOUtils.readEncoding(tsTableBuffer);
+        ReadWriteIOUtils.readCompressionType(tsTableBuffer);
+      }
+
+      ReadWriteIOUtils.readMap(tsTableBuffer); // Column props
+
+      measurementSchemas.add(new MeasurementSchema(columnName, dataType));
+      columnTypes.add(category.toTsFileColumnType());
+    }
+
+    ReadWriteIOUtils.readMap(tsTableBuffer); // Table props (skip)
+
+    return new TableSchema(tableName, measurementSchemas, columnTypes);
+  }
+
+  /**
+   * Fast skip column data without parsing. Critical for performance with many columns.
+   *
+   * @param buffer ByteBuffer to skip data from
+   * @param category Column category to determine how much data to skip
+   */
+  private static void skipColumnData(
+      final ByteBuffer buffer, final TsTableColumnCategory category) {
+    // Skip column name
+    ReadWriteIOUtils.readString(buffer);
+    // Skip data type
+    ReadWriteIOUtils.readDataType(buffer);
+    // Skip encoding and compression for FIELD columns
+    if (category == TsTableColumnCategory.FIELD) {
+      ReadWriteIOUtils.readEncoding(buffer);
+      ReadWriteIOUtils.readCompressionType(buffer);
+    }
+    // Skip column props
+    ReadWriteIOUtils.readMap(buffer);
```

Comment:
May add skipString and skipMap, which only change the buffer position instead of creating temporary objects.